### PR TITLE
Add [options] param.

### PR DESCRIPTION
**Pull request by irfaan**
Options param added to captureTextIntent + captureSpeechIntent calls.

Docs would also need to be updated accordingly.

### CODE REVIEW

File: lib/wit.js [hunk 1]
```diff
@@ -29,24 +29,38 @@ var getHeaders = function (access_token, others) {
...
- * @param [context] json object to be passed to wit for context
+ * @param [options] json object to be passed to wit. Can include any of 'context', 'verbose', 'n'
  * @param callback callback that takes 2 arguments err and the response body
  */
-var captureTextIntent = function (access_token, text, context, callback) {
-    if( !callback) {
-        callback = context;
-        context=undefined;
+    var captureTextIntent = function (access_token, text, options, callback) {
+    if(!callback || typeof callback != "function") {
```
Comment (oliviervaussy) [inline lib/wit.js]: no need to check if it's a function here, if it's not the call will fail as it was not called correctly, that's simplest.

File: lib/wit.js [hunk 2]
```diff
@@ -29,24 +29,38 @@ var getHeaders = function (access_token, others) {
...
+        callback = options;
+        options = false;
```
Comment (oliviervaussy) [inline lib/wit.js]: undefined is fine, it's falsy in javascript

File: lib/wit.js [hunk 3]
```diff
@@ -29,24 +29,38 @@ var getHeaders = function (access_token, others) {
...
     }
-    query_params = {'q': text, 'context': context};
 
-    var options = {
+// Set up the query
+    query_params = {'q': text};
+
+// Add options if passed
+    var allowed_options = ['context', 'verbose', 'n'];
```
Comment (oliviervaussy) [inline lib/wit.js]: let's make it simple and merge those options to query_params, so that new params will be automatically managed and if the param is not managed then Wit will ignore it.

**Comment by oliviervaussy** [comment]
Can we add a quick test for this ? With the current test it should be pretty easy to check if the options passed are passed as query params as well,

Otherwise it looks good thanks for the diff

**Comment by irfaan** [comment]
Sorry - not familiar with testing. I took a look, but from what I understand, you're intercepting the http call and responding with a pre-recorded wit_response, and matching it against wit_response for the test. Is that correct?

Also, how do I check if the options were passed as query params if the query params don't form part of the actual http response for those calls?

File: lib/wit.js [hunk 4]
```diff
@@ -29,24 +29,35 @@ var getHeaders = function (access_token, others) {
...
     }
-    query_params = {'q': text, 'context': context};
 
-    var options = {
+// Set up the query
+    query_params = {'q': text};
+
+// Add options if passed
+    if(options){
+        _.each(options, function(value, key){
```
Comment (oliviervaussy) [inline lib/wit.js:46]: why not `query_params = _.extend(options, {'q':text});` instead of the _each ?

**Comment by oliviervaussy** [comment]
For the test something like that should work :+1: 

``` node
        it('should send options as given in parameter', function (done) {
            var scope = nock('https://api.wit.ai/', {
                reqheaders: {
                    'Authorization': 'Bearer 1234',
                    'Accept': 'application/vnd.wit.20150306'
                }
            }).get('/message?q=set%20alarm%20tomorrow%20at%205pm&context%5Bverbose%5D=true&context%5Bcontext%5D%5Btest%5D=1')
                .reply(200, wit_response);
            wit.captureTextIntent("1234", "set alarm tomorrow at 5pm", {"verbose": true, "context": {"test" : "1"}},function (err, res) {
                assert.isNull(err, "error should be undefined");
                assert.deepEqual(res, wit_response);
                scope.done();
                done();
            });
        });
```

The assertion is done using nock, here we assert that the url for the request is the one defined in the .get('<URL>') function, so if you add the options it will check that your code is actually adding those options to the request. You can of course add different test: test with undefined options, test with empty object options etc... But the basic test should be something like the one above. 

You can add the same kind of test for the speech function as well.

After that your diff will be good to merge, thanks a lot for your help

**Comment by Catacola** [comment]
Merged and updated by hand. These changes along with tests are now in 2.0.0 and on npm. Closing.

**Comment by irfaan** [comment]
Whoops! Sorry for the delay! Meant to make those changes + write the test. Thanks @oliviervaussy @Catacola (esp. appreciate the guidance @oliviervaussy).

**Comment by Catacola** [comment]
It's no problem. Thanks for making this upgrade!
